docs/pages/versions/unversioned/sdk/cellular.md

```diff
@@ -0,0 +1,149 @@
+
+## API
+
+```js
+import * as Cellular from 'expo-cellular';
+```
+
+### Constants
+
+- [`Cellular.allowsVoip`](#cellularallowsvoip)
+- [`Cellular.carrier`](#cellularcarrier)
+- [`Cellular.isoCountryCode`](#cellularisocountrycode)
+- [`Cellular.mobileCountryCode`](#cellularmobilecountrycode)
+- [`Cellular.mobileNetworkCode`](#cellularmobilenetworkcode)
+
+### Methods
+
+- [`Cellular.getCellularGenerationAsync()`](#cellulargetcellulargenerationasync)
+
+### Enum Types
+
+- [`Cellular.CellularGeneration`](#cellularcellulargeneration)
+
+### Errors
+
+- [Error Codes](#error-codes)
+
+## Constants
+
+### `Cellular.allowsVoip`
+
+Indicates if the carrier allows making VoIP calls on its network. On Android, this checks whether the system supports SIP-based VoIP API. See [here](https://developer.android.com/reference/android/net/sip/SipManager.html#isVoipSupported(android.content.Context)) to view more information. 
+
+On iOS, if you configure a device for a carrier and then remove the SIM card, this property retains the `boolean` value indicating the carrier’s policy regarding VoIP. If you then install a new SIM card, its VoIP policy `boolean` replaces the previous value of this property.
+
+On web, this returns `null`.
+
+#### Examples
+
+```js
+Cellular.allowsVoip; // true or false
+```
+
+### `Cellular.carrier`
+
+The name of the user’s home cellular service provider. If the device has duel sim card, only the current active sim card on the network will be returned. On Android, this value is only available when SIM state is [`SIM_STATE_READY`](https://developer.android.com/reference/android/telephony/TelephonyManager.html#SIM_STATE_READY). Otherwise, this returns `null`. 
+
+On iOS, if you configure a device for a carrier and then remove the SIM card, this property retains the name of the carrier. If you then install a new SIM card, its carrier name replaces the previous value of this property. The value for this property is `null` if the user never configured a carrier for the device.
+
+On web, this returns `null`.
+
+#### Examples
+
+```js
+Cellular.carrier; // "T-Mobile" or "Verizon"
+```
+
+### `Cellular.isoCountryCode`
+
+The ISO country code for the user’s cellular service provider. On iOS, the value is `null` if any of the following apply:
+
+* The device is in airplane mode.
+* There is no SIM card in the device.
+* The device is outside of cellular service range.
+
+On web, this returns `null`.
+
+#### Examples
+
+```js
+Cellular.isoCountryCode; // "us" or "au"
+```
+
+### `Cellular.mobileCountryCode`
+
+The mobile country code (MCC) for the user’s current registered cellular service provider. On Android, this value is only available when SIM state is [`SIM_STATE_READY`](https://developer.android.com/reference/android/telephony/TelephonyManager.html#SIM_STATE_READY). Otherwise, this returns `null`. On iOS, the value may be `null` on hardware prior to iPhone 4S when in airplane mode.
```

Comment:
one nuance here -- the value is only available when the SIM state is `SIM_STATE_READY` **at the time the app boots up**, since that's when `getConstants` is called. in practice i don't think it's a huge deal as it looks like the SIM state should almost always be SIM_STATE_READY, but worth documenting at least.